File: 26.remove-duplicates-from-sorted-array.js
/*
 * @lc app=leetcode id=26 lang=javascript
 *
 * [26] Remove Duplicates from Sorted Array
 */

// @lc code=start
/**
 * @param {number[]} nums
 * @return {number}
 */
var removeDuplicates = function(nums) {
  if (nums.length === 0) return nums
  let prev = nums[0]
  let counter = nums.length
  for (let i = 1; i < nums.length; i++) {
    if (prev === nums[i]) {
      nums.splice(i, 1)
      counter--
      i--
      continue
    } else {
      prev = nums[i]
    }
  }
  return counter
};
// @lc code=end

File: cpp/26.remove-duplicates-from-sorted-array.cpp
/*
 * @lc app=leetcode id=26 lang=cpp
 *
 * [26] Remove Duplicates from Sorted Array
 */

// @lc code=start
class Solution
{
public:
    int removeDuplicates(vector<int> &nums)
    {
        // keep track of curr val
        // iterate and push_front when curr val is
        // not equal to next val
        if (nums.size() < 2)
            return nums.size();
        int currVal = nums[0];
        int uniqNums = 1;
        for (int i = 1; i < nums.size(); i++)
        {
            if (currVal != nums[i])
            {
                currVal = nums[i];
                uniqNums++;
            }
            else
            {
                nums.erase(nums.begin() + i);
                i--;
            }
        }
        return uniqNums;
    }
};
// @lc code=end
